src/sagemaker/user_agent.py

```diff
@@ -25,9 +25,23 @@
 PYTHON_VERSION = '{}.{}.{}'.format(sys.version_info.major, sys.version_info.minor, sys.version_info.micro)
 
 
-def prepend_user_agent(client):
+def determine_prefix():
     prefix = 'AWS-SageMaker-Python-SDK/{} Python/{} {}/{} Boto3/{} Botocore/{}'\
         .format(SDK_VERSION, PYTHON_VERSION, OS_NAME, OS_VERSION, boto3.__version__, botocore.__version__)
+
+    try:
+        with open('/etc/opt/ml/sagemaker-notebook-instance-version.txt') as sagemaker_nbi_file:
```

Comment:
Add tests.

Reply:
Yep, I can do that.  Do note that prior to this there weren't any tests on the user_agent.py class, I'll do what I can.